src/backend/expungeservice/expunger/analyzers/time_analyzer.py

```diff
@@ -60,11 +62,37 @@ def _mark_all_acquittals_ineligible_using_mrd_date(expunger):
 
     @staticmethod
     def _evaluate_class_b_felonies(expunger):
-        if expunger.most_recent_charge and expunger.most_recent_charge.disposition.date > TimeAnalyzer.TWENTY_YEARS_AGO:
-            for charge in expunger.class_b_felonies:
-                charge.set_time_ineligible('Time-ineligible under 137.225(5)(a)(A)(i)',
-                                           expunger.most_recent_charge.disposition.date + relativedelta(
-                                               years=TimeAnalyzer.TWENTY_YEARS))
+        for b_felony in expunger.class_b_felonies:
+
+            # if the B felony is acquitted, it is affected the other time eligibility rules,
+            # so leave it untouched here.
+            # if it's convicted, this restriction overrides any of those rules.
+            if not b_felony.acquitted():
+                has_subsequent_charge = False
+                for other_charge in expunger.charges:
+
+                    #if it's an aquittal, the arrest date is what matters
+                    if other_charge.acquitted():
+                        date_of_charge = other_charge.date
+                    #if it's a conviction, the conviction date is what matters
+                    else:
+                        date_of_charge = other_charge.disposition.date
+
+                    if date_of_charge > b_felony.disposition.date:
+                        has_subsequent_charge = True
+                        break
```

Comment:
We can extract a method here: `def calculate_has_subsequent_charge(...) -> bool`